Python/pystate.c

```diff
@@ -1408,7 +1424,12 @@ free_threadstate(_PyThreadStateImpl *tstate)
                sizeof(*tstate));
     }
     else {
+#ifdef HAVE_THREAD_LOCAL
+        // Maximum of 100 thread states in the freelist
```

Comment:
Sorry about the silly question, but where does this limit of 100 thread states come from?

Reply:
I sort of picked it off the top of my head, really. The max freelist limits are [pretty random](https://github.com/python/cpython/blob/85799f1ffd5f285ef93a608b0aaf6acbb464ff9d/Include/internal/pycore_freelist_state.h) (floats use 100, FWIW).